File: data/gencode/ExampleCfg.go
package gencode

import (
	"encoding/json"
	"gs/define"
	"io/ioutil"
	"os"
	"sync"
)

type ExampleCfg struct {
	ExampleSlc []*ExampleInfo `json:"Example"`
	ExampleMap map[int32]*ExampleInfo
	AbbSlc     []*AbbInfo `json:"Abb"`
	AbbMap     map[int32]*AbbInfo
}

type ExampleInfo struct {
	ID         int32            // 数字
	Name       string           // 字符串
	Slc1       []int32          // 数组
	Slc2       []float32        // 数组
	DoubleSlc1 [][]int32        // 二维数组
	DoubleSlc2 [][]string       // 二维数组
	IsBool     bool             // 布尔
	Map1       map[int32]int32  // map类型
	Map2       map[int32]string // map类型
}

type AbbInfo struct {
	ID  int32  // ID
	Sth string // 参数1
}

var exampleCfg *ExampleCfg
var exampleOnce sync.Once

func GetExampleCfg() *ExampleCfg {
	exampleOnce.Do(func() {
		exampleCfg = new(ExampleCfg)
		exampleCfg.init()
	})
	return exampleCfg
}

func (this *ExampleCfg) init() {
	this.ExampleMap = make(map[int32]*ExampleInfo)
	this.AbbMap = make(map[int32]*AbbInfo)
	rootPath := os.Getenv(define.EnvName)
	filePtr, err := os.Open(rootPath + "/data/json/Example.json")
	if err != nil {
		panic(err)
	}
	defer filePtr.Close()
	data, err := ioutil.ReadAll(filePtr)
	if err != nil {
		panic(err)
	}
	err = json.Unmarshal(data, this)
	if err != nil {
		panic(err)
	}
	for _, v := range this.ExampleSlc {
		this.ExampleMap[v.ID] = v
	}
	for _, v := range this.AbbSlc {
		this.AbbMap[v.ID] = v
	}
}

func (this *ExampleCfg) GetExampleById(id int32) (*ExampleInfo, bool) {
	if ret, ok := this.ExampleMap[id]; ok {
		return ret, ok
	}
	return nil, false
}

func (this *ExampleCfg) GetAbbById(id int32) (*AbbInfo, bool) {
	if ret, ok := this.AbbMap[id]; ok {
		return ret, ok
	}
	return nil, false
}

File: data/gencode/ExampleCfg.ts
import Res from "../../common/util/Res"

export class ExampleInfo {
    public ID: number// 数字
    public Name: string// 字符串
    public Slc1: Array<number>// 数组
    public Slc2: Array<number>// 数组
    public DoubleSlc1: Array<Array<number>>// 二维数组
    public DoubleSlc2: Array<Array<string>>// 二维数组
    public IsBool: boolean// 布尔
    public Map1: Map<number,number>// map类型
    public Map2: Map<number,string>// map类型
}

export class AbbInfo {
    public ID: number// ID
    public Sth: string// 参数1
}

export class ExampleConfig {
    private static instance: ExampleConfig
    public static Get(): ExampleConfig {
        if (ExampleConfig.instance == null) {
            ExampleConfig.instance = new ExampleConfig()
            ExampleConfig.instance.init()
        }
        return ExampleConfig.instance
    }

    public ExampleSlc: Array<ExampleInfo>
    public ExampleMap: Map<number, ExampleInfo>

    public AbbSlc: Array<AbbInfo>
    public AbbMap: Map<number, AbbInfo>

    private init(): void {
        this.ExampleSlc = new Array<ExampleInfo>()
        this.ExampleMap = new Map<number, ExampleInfo>()

        this.AbbSlc = new Array<AbbInfo>()
        this.AbbMap = new Map<number, AbbInfo>()

        let jsonData = Res.get<cc.JsonAsset>("json/Example", cc.JsonAsset)
        this.ExampleSlc = jsonData.json['Example']
        this.ExampleSlc.forEach(example => {
            this.ExampleMap.set(example.ID, example)
        })

        this.AbbSlc = jsonData.json['Abb']
        this.AbbSlc.forEach(abb => {
            this.AbbMap.set(abb.ID, abb)
        })

    }
}
